og_migrate/og_migrate.module

```diff
@@ -19,10 +19,21 @@ use Drupal\migrate\Plugin\MigrationInterface;
  * @throws \Drupal\migrate\MigrateSkipRowException
  */
 function og_migrate_migrate_prepare_row(Row $row, MigrateSourceInterface $source, MigrationInterface $migration) {
-  if (in_array($migration->id(), ['d7_field', 'd7_field_instance'])) {
+  // use the source plugin id instead of the migration id,
+  // as the migration id can be altered when using the migration_upgrade module
+  $plugin_id = $source->getPluginId();
+
+  if (in_array($plugin_id, ['d7_field', 'd7_field_instance'])) {
     $field_names = ['og_group_ref', 'og_user_node'];
     if (in_array($row->getSourceProperty('field_name'), $field_names)) {
       throw new MigrateSkipRowException('og-related fields will be migrated separately');
     }
   }
+
+  if (in_array($plugin_id, ['d7_field_instance', 'd7_field_instance_per_view_mode', 'd7_field_instance_per_form_display'])) {
```

Comment:
Neat. I didn't realize these migration plugins existed. Are they in core

We probably need to add an assertion for these migrations. Maybe asserting that the display and form mode configuration entities exist by id?

Reply:
I think that we should assert that the EntityFormDisplay and EntityViewDisplay configuration exists by the appropriate ID, which is probably something like entity_form_display.ENTITY_TYPE.BUNDLE, right?

Reply:
Sorry @mradcliffe, I'm afraid I still don't follow.
However, I suggest I add a test case that checks that fields attached to the default membership type are correctly migrated from the old _og_membership_type_default_ bundle to the new _default_ one. Does that make sense?

Reply:
I am actually not certain this is necessary, as the d7_og_membership_type plugin already maps the og_membership_type_default to default.
So if a migration does have a default membership type, it will be created.
I will mark this as resolved, but feel free to open it again.